### ISSUE TITLE
`tempel-expand` does not expand when the template key contains non-word characters

### ISSUE THREAD

**Issue by hcpsilva** [minad/tempel#47]
When trying to create `org-tempo` styled templates (i.e. prefixed with the character '<'), I've noticed that `tempel-expand` is unable to expand when the template key contains such character.

```lisp
;; -*- lisp -*-

org-mode

(<el "#+begin_src emacs-lisp :tangle yes" n> r> n> "#+end_src"
     :post (progn (tempel-done) (org-edit-src-code)))
(<sh "#+begin_src bash :results output :exports both" n> r> n> "#+end_src"
     :post (progn (tempel-done) (org-edit-src-code)))
(<r "#+begin_src R :session :results output :exports both" n> r> n> "#+end_src"
    :post (progn (tempel-done) (org-edit-src-code)))
(<rg "#+header: :unit cm :width 9 :height 6" n
     "#+begin_src R :session :results output file graphics "
     ":file " (org-babel-temp-file "figure" ".pdf") " "
     ":exports both" n> r> n> "#+end_src"
     :post (progn (tempel-done) (org-edit-src-code)))

(<n "#+name: " p)
(<t "#+title: " p)
```

Here's my Org mode snippets file, and none of these work. I find this problematic given that `org-tempo` is able to use such keys using common `tempo` defines, and I'd also like to maintain the style set by these default Org snippets on my own snippets.

I'd love to poke around and help to find a solution, but my Emacs-fu is young and, so, I still don't grasp most common practices when writing Emacs lisp.

Anyways, thanks for creating such a lovely package that modernizes the reliable and built-in `tempo`!

**Comment by minad** [comment]
We could introduce a configurable prefix character such that the Tempel completion at point functions would know how far to look back. Currently I use `(thing-at-point 'symbol)`. If you place the point somewhere on `key` in a buffer with content `prefix<key`, the function will return only `"key"`. You can try this with `M-:`. But I must say that the choice of `<` as trigger character in org-tempo is not a good one given that it conflicts with electric pair mode or smartparens.

A possible configuration could look like this:

```lisp
;; -*- lisp -*-

org-mode :trigger ?<

(n "#+name: " p)
(t "#+title: " p)
```

I would appreciate a PR which adds support for this. See https://github.com/minad/tempel#contributions.

**Comment by minad** [comment]
@hcpsilva Please let me know if https://github.com/minad/tempel/commit/4da7ea7f40c3da644f31a2b3a78708c52087486a fixes your issue satisfactorily. I took a different (and I think better) approach with the introduction of `tempel-trigger-prefix`.

**Comment by hcpsilva** [comment]
Hi @minad ! Yes, the linked commit did resolve the issue in a very satisfactory manner. I do also believe that the introduction of such custom variable proved to be a better approach, at least on my perspective.

I did try to implement the fine-grained approach, but I gave up somewhat quickly as, in order to pass around such prefix to the already implemented functions, I would have to alter the data structures considerably (add a cons cell to each trigger so it includes the optional prefix, or something like that).

Anyways, thank you very much for resolving this so quickly!
